lib/strategies/ie9.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
```

Comment:
Quite a bit changed actually. 

* Since IE9 doesn't give you the input value until after the key press event completes, I had to [write logic that uses the correct value for formatting](https://github.com/braintree/restricted-input/pull/15/files#diff-acac9442617c62044b93e6e64e18d529R48) and [pads the selection so the cursor ends up in the right spot](https://github.com/braintree/restricted-input/pull/15/files#diff-acac9442617c62044b93e6e64e18d529R54)
* Because of this weirdness, we have to `event.preventDefault()` to make this work correctly
* [Pasting has all kinds of weird behavior](https://github.com/braintree/restricted-input/pull/15/files#diff-acac9442617c62044b93e6e64e18d529R68) that needed to be accounted for
* There's some similar behavior to the iOS strategy, but it's just different enough that it made trying to share behavior. It made more sense to duplicate a bit of the logic, rather than trying to work around each strategies quirks. I did pull out the `getStateToFormat` method from iOS and put it in the Base Strategy.